tests/app/conftest.py

```diff
@@ -138,46 +141,52 @@ def sample_service(notify_db,
     service = Service.query.filter_by(name=service_name).first()
     if not service:
         service = Service(**data)
-        dao_create_service(service, user)
+        dao_create_service(service, user, service_id)
     else:
         if user not in service.users:
             dao_add_user_to_service(service, user)
     return service
 
 
 @pytest.fixture(scope='function')
-def sample_template(notify_db,
-                    notify_db_session,
-                    template_name="Template Name",
-                    template_type="sms",
-                    content="This is a template:\nwith a newline",
-                    archived=False,
-                    subject_line='Subject',
-                    user=None,
-                    service=None,
-                    created_by=None,
-                    process_type='normal'):
+def sample_template(
+    notify_db,
+    notify_db_session,
+    template_name="Template Name",
+    template_type="sms",
+    content="This is a template:\nwith a newline",
+    archived=False,
+    subject_line='Subject',
+    user=None,
+    service=None,
+    created_by=None,
+    process_type='normal',
+    template_id=None
+):
     if user is None:
         user = create_user()
     if service is None:
         service = sample_service(notify_db, notify_db_session)
     if created_by is None:
         created_by = create_user()
-    data = {
-        'name': template_name,
-        'template_type': template_type,
-        'content': content,
-        'service': service,
-        'created_by': created_by,
-        'archived': archived,
-        'process_type': process_type
-    }
-    if template_type in ['email', 'letter']:
-        data.update({
-            'subject': subject_line
-        })
-    template = Template(**data)
-    dao_create_template(template)
+
+    template = Template.query.filter_by(id=template_id).first()
```

Comment:
* the use case of "create a template for a test with a specific ID" is limited, as far as I can think, to specifically tests involving either this or the notify templates (eg 2FA, invite email, reset password email) - and the notify 2FA has its own fixture at sms_code_template
* these functions are already immensely complex, i've been trying to cut them down with my `db.py` stuff anyway
* given that we're calling sample_notification with an ID from only one place, and that's specifically where the test is, I don't think we lose anything by my suggestion #1 of grabbing the template in `_create_slow_delivery_notification`, but we gain keeping slightly complex code limited to only be called in the place where it's actually needed